src/models/Projection.ts

```diff
@@ -251,6 +277,64 @@ export class Projection {
     };
   }
 
+  /** Makes a dictionary from a timerseries to a row so that we can look up the values
+   * based off the date. Eventually would be nice to use this around instead of the
+   * two list scenario we have going right now.
+   */
+  private makeDateDictionary(ts: Timeseries | ActualsTimeseries) {
+    const dict: {
+      [date: string]: CANPredictionTimeseriesRow | CANActualsTimeseriesRow;
+    } = {};
+    ts.forEach((row: CANPredictionTimeseriesRow | CANActualsTimeseriesRow) => {
+      const ts_date = moment(row.date).toString();
+      dict[ts_date] = row;
+    });
+    return dict;
+  }
+
+  private fillDates(summaryWithTimeseries: RegionSummaryWithTimeseries) {
```

Comment:
Okay, this method is pretty nice (fixing up _all_ the timeseries so they align).  Perhaps add a nice comment to complement it?  

I'd also be down to rename to something like alignTimeseriesByDates() or something, but up to you.